spec/kittikrb/cursor_spec.rb

```diff
@@ -122,15 +122,219 @@ def current_rendered_buffer
       end
     end
 
-    describe '#move_to' do
+    context 'move methods' do
       def current_coords
         [cursor.instance_variable_get(:@x), cursor.instance_variable_get(:@y)]
       end
 
-      it 'should properly set absolute position of cursor' do
+      before do
         expect(current_coords).to eq [0, 0]
-        expect(cursor.move_to(5, 10)).to be_instance_of described_class
-        expect(current_coords).to eq [5, 10]
+      end
+
+      describe '#move_to' do
+        it 'should properly set absolute position of cursor' do
+          expect(cursor.move_to(5, 10)).to be_instance_of described_class
+          expect(current_coords).to eq [5, 10]
+        end
+      end
+
+      describe '#move_by' do
+        it 'should properly set offset position of cursor' do
+          cursor.move_to(1, 2)
+          expect(cursor.move_by(3, 7)).to be_instance_of described_class
+          expect(current_coords).to eq [4, 9]
+        end
+      end
+
+      describe '#up' do
+        it 'should properly set up offset position of cursor' do
+          expect(cursor.up).to be_instance_of described_class
+          expect(current_coords).to eq [0, 1]
+        end
+      end
+
+      describe '#down' do
+        it 'should properly set down offset position of cursor' do
+          expect(cursor.down).to be_instance_of described_class
+          expect(current_coords).to eq [0, -1]
+        end
+      end
+
+      describe '#left' do
+        it 'should properly set left offset position of cursor' do
+          expect(cursor.left).to be_instance_of described_class
+          expect(current_coords).to eq [-1, 0]
+        end
+      end
+
+      describe '#right' do
+        it 'should properly set right offset position of cursor' do
+          expect(cursor.right).to be_instance_of described_class
+          expect(current_coords).to eq [1, 0]
+        end
+      end
```

Comment:
Previous 4 tests are very similar and contain too much of repeating code. Please, refactor them. Also, we should be sure that methods working properly with passed arguments. So we need two tests per method.